public/client/TracyLock.hpp

```diff
@@ -14,13 +14,20 @@ namespace tracy
 class LockableCtx
 {
 public:
-    tracy_force_inline LockableCtx( const SourceLocationData* srcloc )
-        : m_id( GetLockCounter().fetch_add( 1, std::memory_order_relaxed ) )
+    tracy_force_inline LockableCtx( const SourceLocationData* srcloc ) :
 #ifdef TRACY_ON_DEMAND
         , m_lockCount( 0 )
         , m_active( false )
+#else
+        m_active(TracyIsStarted)
 #endif
     {
+#if !TRACY_ON_DEMAND
```

Comment:
This is not a valid check if `TRACY_ON_DEMAND` is not defined.

Reply:
You're right, sorry I thought it was defaulting to 0 instead.
I will fix and re-commit.

Reply:
You omitted this one.